donut/modules/bodfeedback/helpers.py

```diff
@@ -0,0 +1,248 @@
+        for e in emails:
+            add_email(complaint_id, e, False)
+    # Add message to database
+    add_msg(complaint_id, data['msg'], data['name'], notification)
+    return complaint_id
+
+
+def add_email(complaint_id, email, notification=True):
+    '''
+    Adds an email to list of addresses subscribed to this complaint
+    returns false if complaint_id is invalid
+    '''
+    if not get_subject(complaint_id): return False
+    query = """
+    INSERT INTO bod_complaint_emails (complaint_id, email)
+    VALUES (%s, %s)
+    """
+    try:
+        with flask.g.pymysql_db.cursor() as cursor:
+            cursor.execute(query, (complaint_id, email))
+        if notification:
+            send_update_email(email, complaint_id)
+    except pymysql.err.IntegrityError:
+        return False
+    return True
+
+
+def remove_email(complaint_id, email):
+    '''
+    Removes 'email' from the list of emails subscribed to this complaint
+    returns False if complaint_id is invalid
+    '''
+    if not get_subject(complaint_id): return False
+    query = 'DELETE FROM bod_complaint_emails WHERE complaint_id = %s AND email = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, (complaint_id, email))
+    return True
+
+
+def add_msg(complaint_id, message, poster, notification=True):
+    '''
+    Adds a message to a complaint in the database
+    and updates status of complaint to 'new_msg'
+    if poster is None or an empty string, it will be replaced with
+    "(anonymous)"
+    If complaint_id is invalid, returns False
+    '''
+    if not get_subject(complaint_id): return False
+    # Add the message
+    query = """
+    INSERT INTO bod_complaint_messages (complaint_id, message, poster, time)
+    VALUES (%s, %s, %s, NOW())
+    """
+    # Update the status to new_msg
+    query2 = 'UPDATE bod_complaint_info SET status = "new_msg" WHERE complaint_id = %s'
+    if not poster:
+        poster = '(anonymous)'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, (complaint_id, message, poster))
+        cursor.execute(query2, complaint_id)
+    if notification:
+        query = 'SELECT email FROM bod_complaint_emails WHERE complaint_id = %s'
+        with flask.g.pymysql_db.cursor() as cursor:
+            cursor.execute(query, complaint_id)
+            res = cursor.fetchall()
+        for row in res:
+            send_update_email(row['email'], complaint_id)
+
+
+def get_link(complaint_id):
+    '''
+    Gets a (fully qualified) link to the view page for this complaint id
+    '''
+    query = 'SELECT uuid FROM bod_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchone()
+        if not res:
+            return None
+        uuid = res['uuid']
+    return flask.url_for(
+        'bodfeedback.bodfeedback_view_complaint', id=uuid, _external=True)
+
+
+def get_id(uuid):
```

Comment:
Why keep a separate `uuid` and `id` for each complaint? Can't the `uuid` just be used as the primary key instead?

Reply:
I can't find an analogous cursor.lastrowid to get the most recently generated UUID (in this line https://github.com/ASCIT/donut/blob/bodfeedback/donut/modules/bodfeedback/helpers.py#L32